### ISSUE TITLE
Refactor cancel modal to avoid prop drilling

### ISSUE THREAD

**Issue by jbalboni** [department-of-veterans-affairs/va.gov-team#29452]
Refactor the CancelAppointmentModal such that:

- It doesn't have the different appointment type conditions (these are not used for homepage refresh)
- Update the component to use connect to Redux directly (using useSelector), rather than needing props to be passed in from the parent component (unless truly necessary).

---

This has now been deployed to the review instance: [http://4eff86bce0676764530f60eddd7998c6.review.vetsgov-internal/](http://4eff86bce0676764530f60eddd7998c6.review.vetsgov-internal/)

**Comment by ldelacosta** [comment]
- Add a review instance link to the ticket @ryanshaw 
-  QA will need to validate the following: 
Canceling a VA booked appointment in VAOS
Canceling a VA request appointment in VAOS
Canceling a Community Care request in VAOS 
Booked an appointment at the VA then canceling the appointment 
Requested a VA appointment, then canceling the request 
Requested a Community Care appointment then canceling the request

**Comment by tlei123** [comment]
@ldelacosta , FYI, testing in RI's still ongoing.

Although i did manage to consistently replicate one bug &mdash; #46298 &mdash; I not totally confident that any other anomalous behaviors legitimately warrants logging a bug-ticket...

TLDR: The actual Cancel-modal itself appears to be working, but after all the pre-existing appts have been canceled, I'm just barely able to book/request an appt so I can cancel it afterward.

The behavior's inconsistent in the RI, and also complicated by the existing appt-limit alert bug #46281 [that one's for Staging but I'm seeing it in the RI too].  That existing bug's making it next to impossible to exactly replicate any possible bug in either environment.

I'll continue testing next Monday afternoon.

cc @ryanshaw

**Comment by tlei123** [comment]
@ryanshaw  , **validation in RI as successful as the environment allowed**.  

Did some further testing today &mdash; managed to find facilities that weren't affected by the appt-limit alert bug above. 
 All Cancellations worked.

Should definitely re-QA everything once in Staging.

Also closed #46298 &mdash; that bug's not front-end and the API-response anomaly found was unique to the associated test-user/appointment.

cc @ldelacosta
